etcd3locker/locker.go

```diff
@@ -0,0 +1,104 @@
+// Tested on etcd 3.1/3.2/3.3
```

Comment:
Would you mind adding some more documentation to the package and the exported types to cover the basics of the etcd locker?

Reply:
Can do.

Reply:
Done in https://github.com/tus/tusd/pull/202/commits/afe6887aed00c0d19e1720011626fe31f728cbad